src/main/java/Duke.java

```diff
@@ -1,10 +1,92 @@
+import java.util.Scanner;
+
 public class Duke {
+    // Assume no more than 100 tasks
+    private static final int MAX_TASKS = 100;
+    private static final Task[] tasks = new Task[MAX_TASKS];
+    private static int taskCount = 0;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner in = new Scanner(System.in);
+        greet();
+        while (executeCommand(in.nextLine()));
     }
+
+    public static void printHorizontalLine() {
+        System.out.println("\t______________________________" +
+                "______________________________");
+    }
+
+    public static void printMessage(String message) {
+        printHorizontalLine();
+        System.out.println(message);
+        printHorizontalLine();
+    }
+
+    public static void list() {
+        printHorizontalLine();
+        System.out.println("\t Here are the tasks in your list:");
+        for (int i = 0; i < taskCount; i++) {
+            System.out.print("\t " + (i + 1) + ". ");
+            System.out.println(tasks[i]);
+        }
+        printHorizontalLine();
+    }
+
+    public static void greet() {
+        printMessage("\t Hello! I'm Duke" + System.lineSeparator() +
+                "\t What can I do for you?");
+    }
+
+    public static void bye() {
+        printMessage("\t Bye. Hope to see you again soon!");
+    }
+
+    public static void addTask(String task) {
+        if (task.startsWith("todo")) {
+            tasks[taskCount] = new Todo(task.replace("todo", "").trim());
+        } else if  (task.startsWith("deadline")) {
+            String[] deadlineInfo = task.replace("deadline", "").split("/by");
+            tasks[taskCount] = new Deadline(deadlineInfo[0].trim(), deadlineInfo[1].trim());
+        } else if (task.startsWith("event")) {
+            String[] eventInfo = task.replace("event", "").split("/at");
+            tasks[taskCount] = new Event(eventInfo[0].trim(), eventInfo[1].trim());
+        } else {
+            System.out.println("No such task.");
+            return;
+        }
+
+        printHorizontalLine();
+        System.out.println("\t Got it. I've added: this task:");
+        System.out.println("\t   " + tasks[taskCount]);
```

Comment:
Perhaps you can put tasks[taskCount++] instead to reduce the code by one line

Reply:
Sure, thanks for pointing out that.